File: 프로그래머스/lv3/12927. 야근 지수/야근 지수.py
import heapq
def solution(n, works):
    answer = 0
    w=[]
    for cur in works:
        w.append(-cur)
    heapq.heapify(w)
    while n and w:
        a=heapq.heappop(w)+1
        if a:
            heapq.heappush(w,a)
        n -= 1
    for cur in w:
        answer+=cur*cur
    return answer

File: 프로그래머스/lv3/12927. 야근 지수/야근 지수.cpp
#include<bits/stdc++.h>
using namespace std;

priority_queue<int>pq;
long long solution(int n, vector<int> works) {
    long long answer = 0;
    for (auto cur : works)
    {
        pq.push(cur);
    }
    while (!pq.empty() && n)
    {
        int top = pq.top()-1;
        pq.pop();
        if (top)
            pq.push(top);
        n -= 1;
    }
    while (!pq.empty()) {
        answer += pow(pq.top(), 2);
        pq.pop();
    }
    return answer;
}